bcipy/gui/gui_main.py

```diff
@@ -196,6 +196,58 @@ def widgets(self) -> List[QWidget]:
         return [self.label_widget, self.help_tip_widget, self.control, self]
 
 
+class IntegerInput(FormInput):
+    """FormInput to select a integer value using a spinbox for selection. Help text is not
+    displayed for spinbox items.
+
+    Parameters:
+    ----------
+        label - form label.
+        value - initial value.
+    """
+
+    def __init__(self, **kwargs):
+        super(IntegerInput, self).__init__(**kwargs)
+
+    def init_control(self, value):
+        """Override FormInput to create a spinbox."""
+        spin_box = QSpinBox()
+        spin_box.setMaximum(100000)
+        return spin_box
+
+    def value(self) -> str:
+        """Override FormInput to return an integer value."""
+        if self.control:
+            return int(self.control.text())
+        return None
+
+
+class FloatInput(FormInput):
+    """FormInput to select a float value using a spinbox for selection. Help text is not
+    displayed for spinbox items.
+
+    Parameters:
+    ----------
+        label - form label.
+        value - initial value.
+    """
+
+    def __init__(self, **kwargs):
+        super(FloatInput, self).__init__(**kwargs)
+
+    def init_control(self, value):
+        """Override FormInput to create a spinbox."""
+        spin_box = QDoubleSpinBox()
+        spin_box.setMaximum(100000)
+        return spin_box
+
+    def value(self) -> float:
+        """Override FormInput to return as a float."""
+        if self.control:
+            return float(self.control.text())
+        return None
```

Comment:
The parent class has the type signature `def value(self) -> str`, and there are at least two places in the code that depend on a `str` return type and will likely throw runtime errors: `FormInput#matches` method and the `ParamsForm#update_parameters` method.

Do you need the values cast to their types? We may need to create an additional method in FormInput with a return type of `Any`, so we have (`cast_value -> Any`, `value -> str`) or (`value -> Any`, `str_value -> str`). I would still like the widget itself to know how to cast its value to a str, since there otherwise ends up being quite a bit of duplication if we have the downstream user do it.